### PR TITLE
Add stateful PBT for NGT service

### PR DESCRIPTION
**Pull request by rinx**
Signed-off-by: Rintaro Okamura <rintaro.okamura@gmail.com>

### Description:

A stateful property-based test for NGT service is introduced.
This test will be useful for finding bugs that depends on the state of the service.

Ref.)
- https://github.com/leanovate/gopter
- https://ymotongpoo.hatenablog.com/entry/2020/12/19/225831

To run it individually,
```sh
go test ./pkg/agent/core/ngt/service \
    -run="TestStatefulNGT" \
    -v \
    -pbt-seed=0 \
    -pbt-min-successful-tests=10 \
    -pbt-enable-application-log
```

### Related Issue:

Nothing

### How Has This Been Tested?:

Nothing

### Environment:

- Go Version: 1.16.6
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.3

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [X] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [X] Have you written new tests for your core changes, as applicable?
- [X] Have you successfully ran tests with your changes locally?

### Checklist:

- [X] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [X] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [X] I have added tests and benchmarks to cover my changes.
- [X] I have ensured all new and existing tests passed.
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pkg/agent/core/ngt/service/ngt_stateful_test.go [hunk 1]
```diff
@@ -0,0 +1,604 @@
...
+	}
+
+	uuidGen = gen.AlphaString().SuchThat(func(s string) bool {
+		return len(s) > 0
+	})
+
+	f32sliceGen = gen.SliceOfN(dimension, gen.Float32())
+)
+
+func init() {
```
[the remark this answers is not in the document]
Comment (rinx) [reply pkg/agent/core/ngt/service/ngt_stateful_test.go:62]: I don't know but, is it an anti-pattern even if it is a test file?

File: pkg/agent/core/ngt/service/ngt_stateful_test.go [hunk 2]
```diff
@@ -0,0 +1,604 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package service manages the main logic of server.
+package service
```
[the remark this answers is not in the document]
Comment (rinx) [reply pkg/agent/core/ngt/service/ngt_stateful_test.go:18]: should I follow this?

**Review by kpango — APPROVED** [review]
LGTM

**Review by hlts2 — APPROVED** [review]
LGTM

**Pull request merged**